sdk/core/azure-core-test/inc/azure/core/test/interceptor_manager.hpp

```diff
@@ -32,6 +33,30 @@
 
 namespace Azure { namespace Core { namespace Test {
 
+  /**
+   * @brief Client Certificate Credential authenticates with the Azure services using a
+   * Tenant ID, Client ID and a client secret.
+   *
+   */
+  class TestClientSecretCredential final : public Core::Credentials::TokenCredential {
```

Comment:
Should we call this something more descriptive, `TestNonExpiringCredential`.  Do the values of environment variables Client / Secret matter or need to exist since this is mocked?

Reply:
From the test point of view, the env vars must always be present to run the tests.

Tests should not know and worry about how they are running (live, playback, etc).  The test-mode should be abstracted.  So, a test must always depend on reading env-vars for service-url and any other config ( like credential's input )